### PR TITLE
Add option to load module before anything else #314

### PR DESCRIPTION
**Pull request by anthonyringoet**
As described in #314.

Some notes:
- `-r` option was already in use for `prefix` so took `b`/`before-module` (like existing `logging-module` option). Feel free to request other naming if preferred.
- Requiring the actual module seemed exactly the same as the existing `logging-module`. In an effort to create a function out of it, I abstracted the core part of `util.js's` `requireServerPluginFromPath` to `requireModule`. This leads to some minor differences in textual error output. Behavior of when errors are thrown is still the same though.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
I think we should keep the same option name of Node.js, following the principle of least surprise (we will bump the major).

File: start.js [hunk 1]
```diff
@@ -59,8 +59,16 @@ function stop (message) {
 
 async function runFastify (args) {
   require('dotenv').config()
-
   const opts = parseArgs(args)
+
+  if (opts.beforeModule) {
```
Comment (mcollina) [inline start.js:64]: how would this be handled for the `watch` option?

**Review by mcollina — CHANGES_REQUESTED** [review]
This _might_ not be exactly what @delvedor needs. If I understand correctly, he needs this to load Elastic APM before everything else. 

Specifically, this should be loading only commonjs and not ESM, and possibly done earlier in the process startup (right after argument parsing.

**Comment by anthonyringoet** [comment]
Makes sense and will make it a lot more straightforward if it can move a level up towards the `start` function. Will look into these changes.

File: start.js (hunk 1 above)
Comment (anthonyringoet) [reply start.js:64]: Watch option checks for changes in `process.cwd()` so currently for both this addition and the custom logger module it won't be triggered if it's outside of the current working directory. Should both of these files be watched explicitly as well?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — COMMENTED** [review]
I think the test case about `require` and `fail to require` is missing. 
1. `require` is really done before the `fastify` is running
2. the behavior when `require is fail to load`

The first test is important as it is the actual function is this PR.

**Review by zekth — CHANGES_REQUESTED** [review]
Like @climba03003 said, missing the tests cases of require working and not working.

Otherwise lgtm

**Comment by anthonyringoet** [comment]
Added support for multiple `-r` flags (same as node cli) and added tests.

@climba03003 How would you go about testing if the require(s) are really done before the fastify server is running?

**Comment by climba03003** [comment]
> 
> 
> Added support for multiple `-r` args (same as node cli) and added tests.
> 
> @climba03003 How would you go about testing if the require(s) are really done before the fastify server is running?

Just like your other test case, you can mutate global variables. Then, you can execute `tap` inside the server files.
It can ensure you receive a the things you want in the server.

I think we can expect, if `required files` use `top-level async` method. The order will break? should this be tested or mention?

**Comment by anthonyringoet** [comment]
@climba03003 Like the previous commit?

The top-level async behavior is the same with `node -r ./some-async-file.js` afaik.

**Review by climba03003 — APPROVED** [review]
Good to know that it is the same as `node -r`.

**Comment by mcollina** [comment]
@zekth is this good to land now?

**Review by zekth — APPROVED** [review]
good work!

**Pull request merged**